astropy/table/tests/test_table.py

```diff
@@ -272,12 +272,13 @@ def test_size_mismatch(self, table_types):
             table_types.Table(cols)
 
     def test_name_none(self, table_types):
-        """Column with name=None can init a table IFF names are supplied"""
-        c = table_types.Column(data=[1, 2])
-        table_types.Table([c], names=('c',))
-        with pytest.raises(TypeError):
-            table_types.Table([c])
-
+        """Column with name=None can init a table whether or not names are supplied"""
+        c = table_types.Column(data=[1, 2], name='c')
+        d = table_types.Column(data=[3, 4])
+        t = table_types.Table([c, d], names=(None, 'd'))
+        assert t.colnames == ['c', 'd']
+        t = table_types.Table([c, d])
+        assert t.colnames == ['c', 'col1']
```

Comment:
Can you also test that if a column with no name is passed and no name is explicitly given with ``names=``, then it fails? (it should, right?)